ocaml/xapi/xapi_session.ml

```diff
@@ -472,15 +472,7 @@ let revalidate_external_session ~__context ~session =
 
           (* CP-827: if the user was suspended (disabled,expired,locked-out), then we must destroy the session *)
           let suspended, _ =
-            is_subject_suspended ~__context ~cache:true authenticated_user_sid
-          in
-          let suspended =
-            if suspended then
-              is_subject_suspended ~__context ~cache:false
-                authenticated_user_sid
-              |> fst
-            else
-              suspended
+            is_subject_suspended ~__context ~cache:false authenticated_user_sid
```

Comment:
Are you sure this solves the original bug?
AFAICT the initial query with 'cache:true' found that the account is missing (suspended), and the 2nd query with 'cache:false' said that it is not suspended (so I was guessing that winbind might also be caching something), but I may have misinterpreted the logs.